File: packages/api/server/lib/folder/folder.go
package folder

import "github.com/honerlaw/mentordoc/server/lib/shared"

type Folder struct {
	shared.Entity

	Name           string  `json:"name"`
	OrganizationId string  `json:"organizationId"`
	ParentFolderId *string `json:"parentFolderId"`
}

File: packages/frontend-shared/src/store/model/folder/folder.ts
import {Entity} from "../entity";
import {Exclude, Expose} from "class-transformer";

@Exclude()
export class Folder extends Entity {

    @Expose()
    public name: string;

    @Expose()
    public organizationId: string;

    @Expose()
    public parentFolderId: string | null;

    @Expose()
    public childCount: number;

}